pandas/core/series.py

```diff
@@ -2227,20 +2229,31 @@ def combine(self, other, func, fill_value=np.nan):
         Series.combine_first : Combine Series values, choosing the calling
             Series's values first
         """
+        self_is_ext = is_extension_array_dtype(self.values)
+        if fill_value is None:
+            fill_value = na_value_for_dtype(self.dtype, False)
+
         if isinstance(other, Series):
             new_index = self.index.union(other.index)
             new_name = ops.get_op_result_name(self, other)
-            new_values = np.empty(len(new_index), dtype=self.dtype)
-            for i, idx in enumerate(new_index):
+            new_values = []
+            for idx in new_index:
                 lv = self.get(idx, fill_value)
                 rv = other.get(idx, fill_value)
                 with np.errstate(all='ignore'):
-                    new_values[i] = func(lv, rv)
+                    new_values.append(func(lv, rv))
         else:
             new_index = self.index
             with np.errstate(all='ignore'):
-                new_values = func(self._values, other)
+                new_values = [func(lv, other) for lv in self._values]
             new_name = self.name
+
+        if self_is_ext and not is_categorical_dtype(self.values):
```

Comment:
The `not is_categorical_dtype(self.values)` signals again that we need a way to pass more information on the dtype to `_from_sequence`. xref https://github.com/pandas-dev/pandas/issues/20747, but not for this PR!

Reply:
sounds good to me

Reply:
pls don't use ``self_is_ext`` simply use ``is_extension_type`` and write this out so its readable

```
if is_categorical_dtype(...):
    pass
elif is_extension_type(...):
   ....
```

why is the try/except here? that is very odd

Reply:
I changed the if test as you suggested.

@jreback wrote:
> why is the try/except here? that is very odd

The idea is to first try to coerce to the same type as the extension dtype, and if that doesn't work, just call the regular constructor for the Series.